browser-extension/mv3/tests/rules/response/responseRule.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { Page } from "@playwright/test";
+import { expect, test } from "../../fixtures";
+import { loadRules } from "../../utils";
+import responseRules from "./response_rules.json";
+import testScenarios, { ResponseRuleTestScenarioData } from "./testScenarios";
+import { IBaseTestData } from "../types";
+
+test.describe("Response Rule", () => {
+  testScenarios.forEach((scenario, i) => {
+    test(`${i + 1}. ${scenario.description}`, async ({ appPage, context }) => {
+      await testResponseRule({ appPage, context, ...scenario });
+    });
+  });
+});
+
+const testResponseRule = async (testScenarioData: ResponseRuleTestScenarioData & IBaseTestData) => {
+  const { appPage, context, ruleIds, testPageURL, pageActions } = testScenarioData;
+  const rules = ruleIds.reduce((acc, ruleId) => ({ ...acc, [ruleId]: responseRules[ruleId] }), {});
+  await loadRules(appPage, rules);
+
+  const testPage = await context.newPage();
+
+  await testPage.addInitScript(initScript);
+
+  await testPage.goto(testPageURL, { waitUntil: "domcontentloaded" });
+
+  const expectedResponsePromises = testScenarioData.expectedResponseModifications
+    .map((expectedResponseModification) => {
+      if (expectedResponseModification.serveWithoutHittingServer !== true) {
+        return testPage.waitForResponse(expectedResponseModification.testUrl);
+      }
+      return null;
+    })
+    .filter(Boolean);
+
+  if (pageActions) {
+    await testPage.evaluate(pageActions);
+  }
+
+  await Promise.all(expectedResponsePromises);
+
+  //@ts-ignore
+  const interceptedResponses = await testPage.evaluate(() => window.__getInterceptedResponses());
+
+  for (const expectedResponseModification of testScenarioData.expectedResponseModifications) {
+    const interceptedResponse = interceptedResponses[expectedResponseModification.testUrl];
+    expect(interceptedResponse).toBeDefined();
+    // expect(interceptedResponse.status).toBe(expectedResponseModification.expectedStatusCode ?? 200);
```

Comment:
this should fix the parsing

```suggestion
    expect(interceptedResponse.status).toBe(parseInt(expectedResponseModification.expectedStatusCode as string) ?? 200);
```